test/libsolidity/syntaxTests/conversion/explicit_conversion_event_to_bytes4.sol

```diff
@@ -0,0 +1,9 @@
+contract Test {
+    event MyCustomEvent(uint256);
```

Comment:
> when we change the PR to use externalSignature()

Since we are not using `externalSignature()` anymore per comments above, I assume this comment becomes outdated too?

Reply:
Actually, I'd still add this. `externalSignature()` was the initial motivation but it would still be good to have something that claerly shows how these types are handled. For example I'm not really sure if it will print `bytes memory` or `bytes` like in an error/event definition.

Reply:
Can function be used as event/error parameter? I keep getting `TypeError 3417 Internal or recursive type is not allowed as error parameter type` for declaration like:
```solidity
error MyCustomError(function (string memory) returns (uint));
```

All other cases you mentioned are added.

Reply:
It can but it must be an external function. I.e.
```solidity
error MyCustomError(function (string memory) external returns (uint));
```
Function pointers are internal by default.